core/src/main/scala/org/apache/spark/ui/UIUtils.scala

```diff
@@ -143,14 +143,10 @@ private[spark] object UIUtils extends Logging {
 
   // Yarn has to go through a proxy so the base uri is provided and has to be on all links
   def uiRoot: String = {
-    if (System.getenv("APPLICATION_WEB_PROXY_BASE") != null) {
-      System.getenv("APPLICATION_WEB_PROXY_BASE")
-    } else if (System.getProperty("spark.ui.proxyBase") != null) {
-      System.getProperty("spark.ui.proxyBase")
-    }
-    else {
-      ""
-    }
+    // SPARK-11484 - Use the proxyBase set by the AM, if not found then use env.
```

Comment:
I think this ultimately makes sense, letting Spark-specific stuff take precedence, though it's worth enumerating here or in the JIRA why this is connected to Oozie / uber mode, and why this is right-er than the current behavior.